.github/actions/init-job-matrices/action.yml

```diff
@@ -45,34 +55,38 @@ runs:
         GITHUB_BASE_REF: ${{ github.base_ref }}
         GITHUB_REF_TYPE: ${{ github.ref_type }}
         GITHUB_REF_NAME: ${{ github.ref_name }}
+        GITHUB_TOKEN: ${{ inputs.github_token }}
+        DEFAULT_OS: ubuntu-20.04
+        DEFAULT_PYTHON_VERSION: 3.8.13
       run: |
         set -e
         
-        cat <<EOT >> /tmp/images-build-matrix.json
+        python3 agent_build_refactored/scripts/get_workflow_info.py > /tmp/workflow_info.json
+        IS_MASTER_RUN="$(cat /tmp/workflow_info.json | jq -c '.is_master_run')"
+        echo "is_master_run=${IS_MASTER_RUN}" >> $GITHUB_OUTPUT
+        echo "to_publish=$(cat /tmp/workflow_info.json | jq -c '.to_publish')" >> $GITHUB_OUTPUT
+        echo "is_production=$(cat /tmp/workflow_info.json | jq -c '.is_production')" >> $GITHUB_OUTPUT
+        echo "version=$(cat /tmp/workflow_info.json | jq -c '.version')" >> $GITHUB_OUTPUT
+        
+        mkdir -p /tmp/matrices
+        
+        cat <<EOT | python3 agent_build_refactored/scripts/filter_matrix_jobs.py --is-master-run "${IS_MASTER_RUN}" > /tmp/matrices/agent_image_build_matrix.json
         ${{ inputs.images-to-build }}
         EOT
+        echo "agent_image_build_matrix_json=$(cat /tmp/matrices/agent_image_build_matrix.json)" >> $GITHUB_OUTPUT
```

Comment:
I think this is a litter clearer:
```
echo ${{ inputs.images-to-build }} | python3 agent_build_refactored/... > /tmp/matrices/...
echo "agent_image_build_matrix_json=$(cat /tmp/matrices/...)" >> $GITHUB_OUTPUT
```

If the temp files are not needed later could even do this:
```
echo "agent_image_build_matrix_json=$(echo ${{ inputs.image-to-build }} | python3 agent_build_refactored/...)" >> $GITHUB_OUTPUT
```
That may be less readable so I'll leave it up to you.

This overall comment applies to the other two variables below also.

Reply:
There reason is because that values are multiline string and with `echo ${{ inputs.images-to-build }}`, it just leads to bash syntactic error. And with `echo "${{ inputs.images-to-build }}"` it just removes all double quoted from the string, so it's not a valid JSON anymore.
But thanks for pointing on this again, because it seems that single quote does the trick `echo '${{ inputs.images-to-build }}'` 👍